src/modules/verifiableCredentials/verifiableCredentialsBase.service.ts

```diff
@@ -1,28 +1,54 @@
+import { TypedDataDomain } from '@ethersproject/abstract-signer';
+import { v4 as uuid } from 'uuid';
 import { SignerService } from '../signer';
+import {
+  verifiableCredentialEIP712Types,
+  verifiablePresentationEIP712Types,
+} from './eip712Types';
 import {
   SignedVerifiableCredential,
+  VerifiablePresentation,
   SignVerifiableCredentialOptions,
-  VerifiableCredential,
+  SignVerifiablePresentationOptions,
   VerifyVerifiableCredentialOptions,
   VerifyVerifiableCredentialResults,
+  CreateVerifiableCredentialParams,
+  EWFCredentialSubject,
+  VerifiableCredential,
+  UnsignedVerifiablePresentation,
 } from './verifiableCredentials.types';
 
 export abstract class VerifiableCredentialsServiceBase {
-  abstract prepareIssueCredential: (
+  protected abstract prepareIssueCredential: (
```

Comment:
Maybe my brain isn't working, I don't understand why it would make sense to use `protected` and `abstract` together. `protected` implies an implementation and `abstract` implies no implementation, to me anyways 🤔

Reply:
Abstract is saying that derived class need to implement such class. Default access modifiers is public, so we could use this method outside our class, but we don;t want to.

Reply:
Ah, makes sense, thanks 👍